test/SqlExtension.IntegrationTests/TestUtils.cs

```diff
@@ -0,0 +1,126 @@
+        /// exception, 'false' to throw. If Null is passed in then all exceptions are thrown.</param>
+        /// <returns>The number of rows affected</returns>
+        public static int ExecuteNonQuery(
+            IDbConnection connection,
+            string commandText,
+            Predicate<Exception> catchException = null)
+        {
+            if (connection == null)
+            {
+                throw new ArgumentNullException(nameof(connection));
+            }
+            if (commandText == null)
+            {
+                throw new ArgumentNullException(nameof(commandText));
+            }
+
+            IDbCommand cmd = null;
+            try
+            {
+                cmd = connection.CreateCommand();
+                cmd.CommandText = commandText;
+                cmd.CommandType = CommandType.Text;
+
+                return cmd.ExecuteNonQuery();
+            }
+            catch (Exception ex)
+            {
+                if (catchException == null || !catchException(ex))
+                {
+                    throw;
+                }
+            }
+            finally
+            {
+                if (cmd != null)
+                {
+                    cmd.Dispose();
+                }
+            }
+
+            return 0;
+        }
+
+        /// <summary>
+        /// Creates a IDbCommand and calls ExecuteScalar against the connection.
+        /// </summary>
+        /// <param name="connection">The connection.  This must be opened.</param>
+        /// <param name="commandText">The scalar T-SQL command.</param>
+        /// <param name="catchException">Optional exception handling.  Pass back 'true' to handle the
+        /// exception, 'false' to throw. If Null is passed in then all exceptions are thrown.</param>
+        /// <returns>The scalar result</returns>
+        public static object ExecuteScalar(
+            IDbConnection connection,
+            string commandText,
+            Predicate<Exception> catchException = null)
+        {
+            if (connection == null)
+            {
+                throw new ArgumentNullException(nameof(connection));
+            }
+            if (commandText == null)
+            {
+                throw new ArgumentNullException(nameof(commandText));
+            }
+
+            IDbCommand cmd = null;
+            try
+            {
+                cmd = connection.CreateCommand();
+                cmd.CommandText = commandText;
+                cmd.CommandType = CommandType.Text;
+                return cmd.ExecuteScalar();
+            }
+            catch (Exception ex)
+            {
+                if (catchException == null || !catchException(ex))
+                {
+                    throw;
+                }
+            }
+            finally
+            {
+                if (cmd != null)
+                {
+                    cmd.Dispose();
```

Comment:
Typically it's easier to put the command creation in a using block which will auto-dispose the command after it's done

Reply:
Switched to a using statement